### PR TITLE
Handling ingress traffic into the AKS cluster

### LINKED ISSUE #28: Infrastructure: Install Nginx Ingress Controller in AKS Cluster
Currently, the Public IP resource provisions with the AKS cluster, however traffic from the Public IP does not direct to underlying services and deployments. We need to deploy nginx ingress controller onto AKS cluster.

### Acceptance Criteria
- [ ] Should deploy the nginx ingress controller
- [ ] Should direct traffic to applications deployed in cluster

### PR DESCRIPTION
**Pull request by decodingahmed**

### CODE REVIEW

File: infrastructure/modules/key_vault/main.tf [hunk 1]
```diff
@@ -1,3 +1,7 @@
+locals {
```
Comment (adelechurchill) [inline infrastructure/modules/key_vault/main.tf:1]: Can locals be moved to a `locals.tf` file to keep consistent with the AKS module?

File: infrastructure/modules/traffic_manager/main.tf [hunk 2]
```diff
@@ -0,0 +1,44 @@
+locals {
```
Comment (adelechurchill) [inline infrastructure/modules/traffic_manager/main.tf:1]: Can locals be moved to a `locals.tf` file to keep consistent with the AKS module? (as above)

File: infrastructure/modules/traffic_manager/main.tf [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    protocol                     = "HTTPS"
+    port                         = 443
+    path                         = var.aks_ingress_health_endpoint
+    interval_in_seconds          = local.interval_in_seconds
+    timeout_in_seconds           = local.timeout_in_seconds
+    tolerated_number_of_failures = local.tolerated_number_of_failures
+  }
+
+  tags = {
+    environment = var.environment_name
```
Comment (guypritchard) [inline infrastructure/modules/traffic_manager/main.tf:35]: Ah yes - @decodingahmed, I see you've sorted this.

**Review by guypritchard — APPROVED** [review]
Really nice work figuring this out!  The proof of this pudding is in the eating - so happy to approve this.

File: infrastructure/versions.tf [hunk 4]
```diff
@@ -5,17 +5,24 @@ terraform {
...
     }
+
+    helm = {
+      source  = "helm"
+      version = ">= 2.4.1"
+    }
+
+    kubernetes = {
+      source  = "kubernetes"
+      version = ">= 2.8.0"
```
Comment (yahya130) [inline infrastructure/versions.tf:16]: Is this the latest version of k8's or AKS? I believe the latest version for AKS is 
1.22.6 on azure, of course I might be wrong :)

**Comment by yahya130** [comment]
Great work, thanks for persisting with terraform especially with the AKS/Ingress controller work, looked very difficult to implement :)

**Review by yahya130 — APPROVED** [review]

**Review by keijoraamat — APPROVED** [review]

**Pull request merged**